gt/main.c

```diff
@@ -1409,6 +1410,44 @@ gt_process_unparsed_incoming_pkt(struct acl_search *acl4,
 	rte_pktmbuf_free(pkt);
 }
 
+static void
+return_message(struct gt_instance *instance)
+{
+	int ret;
+	size_t reply_len;
+	const char *reply_msg;
+	struct dynamic_config *dy_conf = get_dy_conf();
+	struct dy_cmd_entry *entry = mb_alloc_entry(&dy_conf->mb);
+
+	if (entry == NULL) {
+		GT_LOG(ERR, "gt: failed to send new lua update return to Dynamic config block at lcore %d\n",
+			dy_conf->lcore_id);
+		return;
```

Comment:
Instead of simply returning here, we should `goto` to a new label (e.g. out) that makes sure that we call `rte_atomic16_inc()` before leaving in order to avoid a deadlock.